### ISSUE TITLE
Unhandled exception in RatScanner 2.8.2

### ISSUE THREAD

**Issue by EthanCoyne** [RatScanner/RatScanner#119]
**Error**
Unhandled exception in RatScanner 2.8.2
```
Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan`1 name)
   at System.IO.FileSystemWatcher.ParseEventBufferAndNotifyForEach(Byte[] buffer, UInt32 numBytes)
   at System.IO.FileSystemWatcher.ReadDirectoryChangesCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading._IOCompletionCallback.IOCompletionCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executio
```
<details>
<summary>Log</summary>

```
[02:17:42.3173158] > [Info]  ----- RatScanner 2.8.2 -----
[02:17:42.3206852] > [Info]  Starting RatScanner...
[02:17:42.3207188] > [Info]  Prewarming name tool tip...
[02:17:42.4247926] > [Info]  Prewarming icon tool tip...
[02:17:42.4440380] > [Info]  Checking for new updates...
[02:17:42.4798247] > [Info]  Loading resource "RSClientVersion"...
[02:17:57.1405181] > [Info]  Loading config...
[02:17:57.1429609] > [Warning] Config version (-1) is not supported!
[02:17:57.1769133] > [Info]  Detected 2560x1440 Resolution.

You can adjust this inside the settings.
[02:17:58.9177321] > [Info]  Loading item data...
[02:17:58.9495250] > [Info]  Loading resource "RSItemDataVersion"...
[02:17:59.0699234] > [Info]  A new item data version is available: 20210630
[02:17:59.0699311] > [Info]  Loading resource "RSItemDataLink"...
[02:17:59.1959091] > [Info]  Downloading file "https://api.ratscanner.com/v3/file/items-en-30.06.2021.json"...
[02:18:01.8382941] > [Info]  Loading progress data...
[02:18:02.0576561] > [Info]  Loading price data...
[02:18:02.3174846] > [Info]  Setting default item...
[02:18:02.3178164] > [Info]  Initializing RatEye...
[02:18:04.4679870] > [Info]  Initializing hotkey manager...
[02:18:04.4695961] > [Info]  Setting up data update routines...
[02:18:04.4696135] > [Info]  Ready!
[02:18:04.4953774] > [Info]  Loading resource "RSPatreonLink"...
[02:18:04.6189471] > [Info]  Loading resource "RSDiscordLink"...
[02:18:04.7460790] > [Info]  Loading resource "RSGithubLink"...
[02:18:39.3634310] > [Info]  Applying config...
[02:18:40.8814343] > [Info]  Saving config...
[02:18:40.8921677] > [Info]  Config saved!
[02:28:04.4733878] > [Info]  Refreshing TarkovTracker DB...
[02:38:04.9970239] > [Info]  Refreshing TarkovTracker DB...
[02:48:05.4216323] > [Info]  Refreshing TarkovTracker DB...
[02:50:47.2204254] > [Error] Unhandled exception in RatScanner 2.8.2
 -------------------- 
 Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan`1 name)
   at System.IO.FileSystemWatcher.ParseEventBufferAndNotifyForEach(Byte[] buffer, UInt32 numBytes)
   at System.IO.FileSystemWatcher.ReadDirectoryChangesCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading._IOCompletionCallback.IOCompletionCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous 
```
</details>

When this exception happens, usually at a trader and an exception happens for each new item that has to load in the trader's inventory as I scroll down

**Comment by hmichaelkim** [comment]
I get this error as well.

**Comment by gunstarpl** [comment]
Having same issue. Was fine during an entire raid after updating, but now keeps crashing constantly while off-raid.

**Comment by gunstarpl** [comment]
**Error**
Unhandled exception in RatScanner 2.8.2
```
Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan`1 name)
   at System.IO.FileSystemWatcher.ParseEventBufferAndNotifyForEach(Byte[] buffer, UInt32 numBytes)
   at System.IO.FileSystemWatcher.ReadDirectoryChangesCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous l
```
<details>
<summary>Log</summary>

```
[06:37:28.3304704] > [Info]  ----- RatScanner 2.8.2 -----
[06:37:28.3349815] > [Info]  Starting RatScanner...
[06:37:28.3350297] > [Info]  Prewarming name tool tip...
[06:37:28.4342130] > [Info]  Prewarming icon tool tip...
[06:37:28.4598702] > [Info]  Checking for new updates...
[06:37:28.4654165] > [Info]  Loading resource "RSClientVersion"...
[06:37:29.0993277] > [Info]  Loading config...
[06:37:29.1079079] > [Info]  Loading item data...
[06:37:29.1085995] > [Info]  Loading resource "RSItemDataVersion"...
[06:37:31.5276637] > [Info]  Loading TarkovTracker...
[06:37:32.6423645] > [Info]  Loading progress data...
[06:37:32.9015394] > [Info]  Loading price data...
[06:37:33.6794651] > [Info]  Setting default item...
[06:37:33.6799270] > [Info]  Initializing RatEye...
[06:37:36.0405319] > [Info]  Initializing hotkey manager...
[06:37:36.0429174] > [Info]  Setting up data update routines...
[06:37:36.0429342] > [Info]  Ready!
[06:37:36.1526393] > [Info]  Loading resource "RSPatreonLink"...
[06:37:36.5291426] > [Info]  Loading resource "RSDiscordLink"...
[06:37:36.9100312] > [Info]  Loading resource "RSGithubLink"...
[06:39:15.5068748] > [Error] Unhandled exception in RatScanner 2.8.2
 -------------------- 
 Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan`1 name)
   at System.IO.FileSystemWatcher.ParseEventBufferAndNotifyForEach(Byte[] buffer, UInt32 numBytes)
   at System.IO.FileSystemWatcher.ReadDirectoryChangesCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pNativeOverlapped)
[06:39:16.5895716] > [Error] Unhandled exception in RatScanner 2.8.2
 -------------------- 
 Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System
```
</details>

**Comment by caffeinewriter** [comment]
I also ran into this, also while off-raid. 

**Error**
Unhandled exception in RatScanner 2.8.2
```
Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatStash.Database.ParseItemCacheHashIndex(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan`1 name)
   at System.IO.FileSystemWatcher.ParseEventBufferAndNotifyForEach(Byte[] buffer)
   at System.IO.FileSystemWatcher.ReadDirectoryChangesCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading.ThreadPoolBoundHandleOverlapped.CompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOv
```
<details>
<summary>Log</summary>

```
[06:43:17.7273026] > [Info]  ----- RatScanner 2.8.2 -----
[06:43:17.7340399] > [Info]  Starting RatScanner...
[06:43:17.7341071] > [Info]  Prewarming name tool tip...
[06:43:17.9779832] > [Info]  Prewarming icon tool tip...
[06:43:18.0466075] > [Info]  Checking for new updates...
[06:43:18.0540567] > [Info]  Loading resource "RSClientVersion"...
[06:43:18.5638966] > [Info]  Loading config...
[06:43:18.5759282] > [Info]  Loading item data...
[06:43:18.5770103] > [Info]  Loading resource "RSItemDataVersion"...
[06:43:18.6218012] > [Info]  Loading progress data...
[06:43:18.8300506] > [Info]  Loading price data...
[06:43:19.3603208] > [Info]  Setting default item...
[06:43:19.3609062] > [Info]  Initializing RatEye...
[06:43:21.1936270] > [Info]  Loading progress data...
[06:43:21.3523804] > [Info]  Loading price data...
[06:43:21.8972351] > [Info]  Setting default item...
[06:43:21.8979779] > [Info]  Initializing RatEye...
[06:43:23.3038790] > [Info]  Initializing hotkey manager...
[06:43:23.3055605] > [Info]  Setting up data update routines...
[06:43:23.3055808] > [Info]  Ready!
[06:43:23.4434400] > [Info]  Loading resource "RSPatreonLink"...
[06:43:23.7525618] > [Info]  Loading resource "RSDiscordLink"...
[06:43:24.0060401] > [Info]  Loading resource "RSGithubLink"...
[06:43:26.0036899] > [Info]  Initializing hotkey manager...
[06:43:26.0060180] > [Info]  Setting up data update routines...
[06:43:26.0060444] > [Info]  Ready!
[06:43:26.1658840] > [Info]  Loading resource "RSPatreonLink"...
[06:43:26.4174650] > [Info]  Loading resource "RSDiscordLink"...
[06:43:26.6652769] > [Info]  Loading resource "RSGithubLink"...
[06:53:23.3196425] > [Info]  Refreshing TarkovTracker DB...
[07:03:23.3395395] > [Info]  Refreshing TarkovTracker DB...
[07:03:53.3533052] > [Error] Unhandled exception in RatScanner 2.8.2
 -------------------- 
 Newtonsoft.Json.JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at RatStash.CacheHashIndexParser.Parse(String filepath)
   at RatStash.Database.ParseItemCacheHashIndex(String filepath)
   at RatEye.IconManager.LoadDynamicCorrelationData()
   at RatEye.IconManager.OnDynamicCorrelationDataChange(Object source, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan`1 name)
   at System.IO.FileSystemWatcher.ParseEventBufferAndNotifyForEach(Byte[] buffer)
   at System.IO.FileSystemWatcher.ReadDirectoryChangesCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading.ThreadPoolBoundHandleOverlapped.CompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped)
   at System.Threading._IOCompletionCallback.IOCompletionCallback_Context(Object state)
   at System.Threading.Exec
```
</details>

**Comment by Blightbuster** [comment]
Thanks for reporting, fixed in v2.8.3
